midas/query.py

```diff
@@ -0,0 +1,59 @@
+from sqlalchemy.ext.declarative.api import DeclarativeMeta
+import sqlalchemy
+
+from .session_creator import session
+
+
+class Query:
+    """
+    Extends Query class of sqlalchemy, this class also supports the query API.
+    """
+    def __init__(self, table_type: DeclarativeMeta, query: sqlalchemy.orm.query.Query):
+        """
+        :param table_type: Terrorist or Organization or Event.
+        :param query: Query object of sqlalchemy.
+        """
+        self.type = table_type
+        self.query = query
+
+    def __getattr__(self, item):
+        returned = getattr(self.query, item)
+        if isinstance(returned, sqlalchemy.orm.query.Query) \
+                or (callable(returned) and isinstance(returned(), sqlalchemy.orm.query.Query)):
```

Comment:
You shouldn't run whatever callable you're given, you might execute a super heavy query twice.
You're also wrapping something that is not a query inside your query, that's dangerous